File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {

    hasDuplicate(nums) {
        let hashSet = new Set ([nums[0]])
        for (let i = 1; i < nums.length;i++){
            if (hashSet.has(nums[i])){
                return true 
                
            }else{ 
                hashSet.add(nums[i])

                }
        }
        return false
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        i=0
        j= len(nums)-1
        output = float("inf")
        while i <=j:
            if nums[i] < nums[j]:
                output=min(output,nums[i])
                break
            m = (i+j)//2

            if nums[i] <= nums[m]:
                output=min(output,nums[m])
                i=m+1
            else:
                output=min(output,nums[m])
                j=m-1
        return output




                
        

        